src/drivers/azure-storage-table.ts

```diff
@@ -0,0 +1,143 @@
+   * @default null
+   */
+  accountName?: string;
+  /**
+   * The name of the table. All entities will be stored in the same table.
+   * @default 'unstorage'
+   */
+  tableName?: string;
+  /**
+   * The partition key. All entities will be stored in the same partition.
+   * @default 'unstorage'
+   */
+  partitionKey?: string;
+  /**
+   * The account key. If provided, the SAS key will be ignored. Only available in Node.js runtime.
+   * @default null
+   */
+  accountKey?: string;
+  /**
+   * The SAS key. If provided, the account key will be ignored.
+   * @default null
+   */
+  sasKey?: string;
+  /**
+   * The connection string. If provided, the account key and SAS key will be ignored. Only available in Node.js runtime.
+   * @default null
+   */
+  connectionString?: string;
+}
+
+export default defineDriver((opts: AzureStorageTableOptions = {}) => {
+  const {
+    accountName = null,
+    tableName = "unstorage",
+    partitionKey = "unstorage",
+    accountKey = null,
+    sasKey = null,
+    connectionString = null,
+  } = opts;
+  if (!accountName)
+    throw new Error(
+      "Account name is required to use the Azure Storage Table driver."
+    );
+
+  let client: TableClient;
+  if (accountKey) {
+    // AzureNamedKeyCredential is only available in Node.js runtime, not in browsers
+    const credential = new AzureNamedKeyCredential(accountName, accountKey);
+    client = new TableClient(
+      `https://${accountName}.table.core.windows.net`,
+      tableName,
+      credential
+    );
+  } else if (sasKey) {
+    const credential = new AzureSASCredential(sasKey);
+    client = new TableClient(
+      `https://${accountName}.table.core.windows.net`,
+      tableName,
+      credential
+    );
+  } else if (connectionString) {
+    // fromConnectionString is only available in Node.js runtime, not in browsers
+    client = TableClient.fromConnectionString(connectionString, tableName);
+  } else {
+    const credential = new DefaultAzureCredential();
+    client = new TableClient(
+      `https://${accountName}.table.core.windows.net`,
+      tableName,
+      credential
+    );
+  }
+
+  return {
+    async hasItem(key) {
+      try {
+        await client.getEntity(partitionKey, key);
+        return true;
+      } catch {
+        return false;
+      }
+    },
+    async getItem(key) {
+      try {
+        const entity = await client.getEntity(partitionKey, key);
+        return entity.unstorageValue;
+      } catch {
+        return null;
+      }
+    },
+    async setItem(key, value) {
+      const entity: TableEntity = {
+        partitionKey,
+        rowKey: key,
+        unstorageValue: value,
+      };
+      await client.upsertEntity(entity, "Replace");
+      return;
+    },
+    async removeItem(key) {
+      await client.deleteEntity(partitionKey, key);
+      return;
+    },
+    async getKeys() {
+      const iterator = client.listEntities().byPage({ maxPageSize: 1000 });
```

Comment:
1000 could be maybe configurable?

Reply:
1000 is just the biggest number possible to reduce the amount of network requests. If you think there is a benefit for the user, I can of course introduce a prameter for that.

Reply:
You are the pro to decide this. From an overall look, seems valid to make it configurable if the client library itself accepts it.